lib/services/api/session_manager.ts

```diff
@@ -451,12 +445,17 @@ export class SNSessionManager extends PureService<SessionEvent> {
   }
 
   private async handleSuccessAuthResponse(
-    response: RegistrationResponse | SignInResponse | ChangePasswordResponse
+    response: RegistrationResponse | SignInResponse | ChangePasswordResponse,
+    rootKey: SNRootKey
   ) {
+    const wrappingKey = await this.challengeService.getWrappingKeyIfApplicable(true);
```

Comment:
Currently the app asks for your passcode before it does anything related to signing in/registering. This has the beneficial side-effect of preventing users from launching a destructive operation without being able to complete it.

What happens if a user can't remember their passcode, or it they close the app without inputting their passcode here?

Reply:
Good point, I'll re-work this.